opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java

```diff
@@ -1317,17 +1323,47 @@ public void setSuggesterConfig(final SuggesterConfig config) {
         this.suggesterConfig = config;
     }
 
-    public StatsdConfig getStatsdConfig() {
+    public BaseStatsdConfig getStatsdConfig() {
         return statsdConfig;
     }
 
-    public void setStatsdConfig(final StatsdConfig config) {
+    public void setStatsdConfig(final BaseStatsdConfig config) {
         if (config == null) {
             throw new IllegalArgumentException("Cannot set Statsd configuration to null");
         }
         this.statsdConfig = config;
     }
 
+    public BaseGraphiteConfig getGraphiteConfig() {
+        return graphiteConfig;
+    }
+
+    public void setGraphiteConfig(BaseGraphiteConfig config) {
+        if (config == null) {
+            throw new IllegalArgumentException("Cannot set Graphite configuration to null");
+        }
+        this.graphiteConfig = config;
+    }
+
+    public MeterRegistryType getWebAppMeterRegistryType() {
+        return webAppMeterRegistryType;
+    }
+
+    public void setWebAppMeterRegistryType(MeterRegistryType registryType) {
+        this.webAppMeterRegistryType = registryType;
+    }
+
+    public MeterRegistryType getIndexerMeterRegistryType() {
+        return indexerMeterRegistryType;
+    }
+
+    public void setIndexerMeterRegistryType(MeterRegistryType registryType) {
+        if (registryType == MeterRegistryType.PROMETHEUS) {
+            throw new IllegalArgumentException("unsupported registry type");
```

Comment:
Can you elaborate on this? It smells off: a brand new enum yet partially illegal with this brand new API

Reply:
Well, the indexer does not have embedded web server so cannot support Prometheus registry. Do you think it would make sense to split `MeterRegistryType` ? Or even `Metrics` ?